### PR TITLE
Add migration cmdlets

### PR DESCRIPTION
**Pull request by DeepakRajendranMsft**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@DeepakRajendranMsft__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by DeepakRajendranMsft** [comment]
@markcowl, CI is good now :)

**Comment by cormacpayne** [comment]
On demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1072/

**Comment by haitch** [comment]
it failed with this test (same test failed in my PR also), someone should fix this test 
Microsoft.WindowsAzure.Commands.Test.Websites.SaveAzureWebsiteLogTests.**SaveAzureWebsiteLogWithNoFileExtensionTest**

File: src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureNetworkSecurityGroup.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+using Microsoft.Azure;
+using Microsoft.WindowsAzure.Management.Network;
+using System.Management.Automation;
+
+namespace Microsoft.WindowsAzure.Commands.ServiceManagement.IaaS.Network
+{
+    /// <summary>
+    /// Migrate ASM Network Security Group to ARM
+    /// </summary>
+    [Cmdlet(VerbsCommon.Move, "AzureNetworkSecurityGroup"), OutputType(typeof(OperationStatusResponse))]
```
Comment (markcowl) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureNetworkSecurityGroup.cs]: SupportsShouldProcess=true

File: src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureNetworkSecurityGroup.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        {
+            get;
+            set;
+        }
+
+        protected override void OnProcessRecord()
+        {
+            ServiceManagementProfile.Initialize();
+
+            if (this.Validate.IsPresent)
```
Comment (markcowl) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureNetworkSecurityGroup.cs]: Protect each of these executions with ShouldProcess

File: src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureReservedIP.cs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Microsoft.WindowsAzure.Management.Network;
+using System.Management.Automation;
+
+namespace Microsoft.WindowsAzure.Commands.ServiceManagement.IaaS.Network
+{
+    /// <summary>
+    /// Migrate ASM ReservedIP to ARM
+    /// </summary>
+    [Cmdlet(VerbsCommon.Move, "AzureReservedIP"), OutputType(typeof(OperationStatusResponse))]
```
Comment (markcowl) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureReservedIP.cs]: SupportsShouldProcess=true

File: src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureReservedIP.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        {
+            get;
+            set;
+        }
+
+        protected override void OnProcessRecord()
+        {
+            ServiceManagementProfile.Initialize();
+
+            if (this.Validate.IsPresent)
```
Comment (markcowl) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureReservedIP.cs]: Protect each of these execution with if (ShouldProcess()) {...}

File: src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureRouteTable.cs [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+using Microsoft.Azure;
+using Microsoft.WindowsAzure.Management.Network;
+using System.Management.Automation;
+
+namespace Microsoft.WindowsAzure.Commands.ServiceManagement.IaaS.Network
+{
+    /// <summary>
+    /// Migrate ASM Route table to ARM
+    /// </summary>
+    [Cmdlet(VerbsCommon.Move, "AzureRouteTable"), OutputType(typeof(OperationStatusResponse))]
```
Comment (markcowl) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureRouteTable.cs]: Same comments on SHouldProcess

File: src/ServiceManagement/Services/Commands/packages.config [hunk 6]
```diff
@@ -14,10 +14,11 @@
   <package id="Microsoft.Net.Http" version="2.2.28" targetFramework="net45" />
   <package id="Microsoft.Rest.ClientRuntime" version="2.3.2" targetFramework="net45" />
   <package id="Microsoft.Rest.ClientRuntime.Azure.Authentication" version="2.2.7-preview" targetFramework="net45" />
+  <package id="Microsoft.Web.Deployment" version="3.5.0" targetFramework="net45" />
```
Comment (markcowl) [inline src/ServiceManagement/Services/Commands/packages.config]: ?? why is this added?

Comment (DeepakRajendranMsft) [inline src/ServiceManagement/Services/Commands/packages.config]: hmm not sure how. will remove it

**Comment by DeepakRajendranMsft** [comment]
@markcowl why do you think we need ShouldProcess() here? we do not need any confirmation of sorts from the user?

File: src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureReservedIP.cs (hunk 4 above)
Comment (DeepakRajendranMsft) [inline src/ServiceManagement/Compute/Commands.ServiceManagement/Migration/MoveAzureReservedIP.cs]: why do we need to protect these executions? this is not like a delete operation and dont need a confirmation for customers.

also, this is very similar to the other Move-* cmdlets (so wouldnt want to introduce breaking change/break uniformity unless required)

File: src/ServiceManagement/Common/Commands.ScenarioTest/Resources/ServiceManagement/ServiceManagementTests.ps1 [hunk 7]
```diff
@@ -826,6 +826,91 @@ function Test-MigrationAbortAzureStorageAccount
...
+    $location = Get-DefaultLocation
+    New-AzureRouteTable -Name $routeTableName -location $location
+
+    # Validate move
+    $status = Move-AzureRouteTable -RouteTableName $routeTableName -Validate
+    Assert-NotNull $status
+    Assert-Null $status.ValidationMessages
+
+    # Prepare move
+    Move-AzureRouteTable -RouteTableName $routeTableName -Prepare -Confirm
```
Comment (markcowl) [inline src/ServiceManagement/Common/Commands.ScenarioTest/Resources/ServiceManagement/ServiceManagementTests.ps1]: Do not pass -Confirm

**Comment by markcowl** [comment]
@DeepakRajendranMsft It looks like you added the -Confirm parameter to each of your cmdlet invocatiosn in script.  -Confirm tells the cmdlet to always prompt on ShouldProcess.  Remove this parameter and these issues should go away

**Comment by cormacpayne** [comment]
On demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1083/

**Comment by DeepakRajendranMsft** [comment]
@markcowl @cormacpayne are we good for the merge?
